typescript/trans/typechecking/typescript.nabl2

```diff
@@ -253,7 +255,36 @@ rules
       Return{"function"} <- function_scope,
       Return{"function"} : returnTy,
       Map1 [[ body ^ (function_scope) ]].
-     
+    
+    
+    [[ ClassDeclaration(None(), _, parent, members) ^ (s) ]] :=
+      new clazz,
+      clazz -P-> s,
+      Map1 [[ members ^ (clazz) ]],
+      ClassParent [[ parent ^ (s): parentTy ]].
+      
+    [[ ClassDeclaration(Some(name), _, parent, members) ^ (s) ]] :=
+      new clazz,
+      clazz -P-> s,
+      Value{name} <- s,
+      classTy == RECORD(clazz),
+      Value{name} : classTy !,
+      Type{name} <- s,
```

Comment:
Classes are also types. Not sure how we can play this out. In the following way, we might allow assignment of a record to a class-type, which should be illegal?